gcp_variant_transforms/testing/integration/run_tests.py

```diff
@@ -0,0 +1,254 @@
+# limitations under the License.
+
+"""Integration testing runner."""
+
+import argparse
+import json
+import multiprocessing
+import os
+
+from datetime import datetime
+from google.cloud import bigquery
+
+from googleapiclient import discovery
+from oauth2client.client import GoogleCredentials
+import time
+
+IMAGE_NAME = 'gcr.io/gcp-variant-transforms/gcp-variant-transforms'
+PIPELINE_NAME = 'gcp-variant-transforms-integration-test'
+SCOPES = ['https://www.googleapis.com/auth/bigquery']
+DEFAULT_ZONES = ['us-west1-b']
+SCRIPT_PATH = '/opt/gcp_variant_transforms/bin/vcf_to_bq'
+
+
+class TestCaseFailure(Exception):
+  """Exception for failed test cases."""
+  pass
+
+
+class TestCase(object):
+  """Test case that holds information to run in pipelines API.
+
+  To define a new integration test case, create a json file in
+  gcp_variant_transforms/testing/integration directory and specify at least
+  test_name, table_name, and input_pattern for the integration test.
+  """
+
+  def __init__(self,
+               context,
+               test_name,
+               table_name,
+               input_pattern,
+               **kwargs):
+
+    self._name = test_name
+    dataset_id = context.dataset_id
+    output_table = '{}:{}.{}'.format(context.project, dataset_id, table_name)
+    args = ['--input_pattern {}'.format(input_pattern),
+            '--output_table {}'.format(output_table),
+            '--project {}'.format(context.project),
+            '--staging_location {}'.format(context.staging_location),
+            '--temp_location {}'.format(context.temp_location),
+            '--job_name {}-{}'.format(test_name, dataset_id.replace('_', '-'))]
+    for k, v in kwargs.iteritems():
+      args.append('--{} {}'.format(k, v))
+
+    self._pipelines_api_request = {
+        'pipelineArgs': {
+            'projectId': context.project,
+            'logging': {'gcsPath': context.logging_location},
+            'serviceAccount': {'scopes': SCOPES}
+        },
+        'ephemeralPipeline': {
+            'projectId': context.project,
+            'name': PIPELINE_NAME,
+            'resources': {'zones': DEFAULT_ZONES},
+            'docker': {
+                'imageName': IMAGE_NAME,
+                'cmd': ' '.join([SCRIPT_PATH] + args)
+            },
+        }
+    }
+
+  def get_name(self):
+    return self._name
+
+  def run(self, context):
+    service = discovery.build(
+        'genomics', 'v1alpha2', credentials=context.credentials)
+    request = service.pipelines().run(body=self._pipelines_api_request)
+
+    operation_name = request.execute()['name']
+    response = self._wait_for_operation_done(service, operation_name)
+
+    _handle_failure(response)
+    return response
+
+  def _wait_for_operation_done(self, service, operation_name):
+    operations = service.operations()
+    request = operations.get(name=operation_name)
+    response = request.execute()
+    while not response['done']:
+      time.sleep(2)
```

Comment:
nit: let's wait a bit longer? like 10 seconds?

Reply:
Since these are REST API requests, it might be safer to have a exponential backoff retry policy.  What do you think?

Reply:
I think in this case since the job not being complete is not a failure we should keep the time constant. However, we should probably increase wait time and initially wait for ~60 seconds since most tests take at least this long to start.